lib/client/doc.js

```diff
@@ -919,6 +920,21 @@ Doc.prototype.resume = function() {
   this.flush();
 };
 
+// Create a snapshot that can be serialized, deserialized, and passed into `Doc.ingestSnapshot`.
+// By default, this implementation defensively clones the document data
+// to guard against mutations. You can opt out of cloning by passing `true`
+// as the optional `doNotClone` argument.
+//
+// @param doNotClone (optional) If `true`, the document data is not cloned. This can be used
```

Comment:
I'm not sure we should let people do this. I know cloning is a performance overhead, but the failure mode is bad if consumers accidentally mutate the snapshot (because they'll mutate `doc.data`, which shouldn't happen outside of OT).

Reply:
The main use case for this is to serialize the snapshot and send it over the network. I'm of the opinion that consumers should be allowed to opt out of cloning if they know for sure that their code will never mutate the data.

Reply:
I still personally think that the risks of weird side-effects massively outweigh the relatively small performance impact of cloning. Cloning is a perfectly normal, defensive thing to do. `json0` itself uses clone in many places through its lifecycle.

(As a side-note, `json0` is just using a [`JSON` approach to cloning](https://github.com/ottypes/json0/blob/90a3ae26364c4fa3b19b6df34dad46707a704421/lib/json0.js#L42-L44), although I believe the comment about it being fast is out-of-date).

Reply:
Got it. That simplifies things. Will do!